dsgrid/config/registration_models.py

```diff
@@ -0,0 +1,129 @@
+"""Contains data models to control registration of test projects and datasets."""
+
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Any, Optional
+
+from pydantic import Field, model_validator
+from typing_extensions import Annotated
+
+from dsgrid.data_models import DSGBaseModel
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.utils.files import load_data
+
+
+class DatasetRegistrationModel(DSGBaseModel):
+    """Defines a dataset to be registered and optionally submitted to a project."""
+
+    dataset_id: Annotated[str, Field(description="Dataset ID")]
+    dataset_path: Annotated[
+        Path, Field(description="Directory containing load_data/load_data_lookup.parquet")
+    ]
+    config_file: Annotated[Path, Field(description="Path to dataset.json5")]
+    dimension_mapping_file: Annotated[
+        Optional[Path],
+        Field(
+            description="Path to file containing mappings of dataset-to-project dimensions",
+            default=None,
+        ),
+    ]
+    dimension_mapping_references_file: Annotated[
+        Optional[Path],
+        Field(
+            description="Path to file containing references to mappings of dataset-to-project dimensions",
+            default=None,
+        ),
+    ]
```

Comment:
Should these fields be associated with a specific project_id?

Reply:
The `ProjectRegistrationModel` contains a list of `DatasetRegistrationModel` instances, so that is implicit. If this registration file defined a dataset in multiple projects, there would be two `DatasetRegistrationModel` instances.